yoti-sdk-impl/src/main/java/com/yoti/api/client/spi/remote/call/docs/DocScanService.java

```diff
@@ -0,0 +1,277 @@
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_GET;
+import static com.yoti.api.client.spi.remote.call.HttpMethod.HTTP_POST;
+import static com.yoti.api.client.spi.remote.call.YotiConstants.*;
+import static com.yoti.api.client.spi.remote.util.Validation.notNull;
+import static com.yoti.api.client.spi.remote.util.Validation.notNullOrEmpty;
+
+import java.io.IOException;
+import java.io.UnsupportedEncodingException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.security.KeyPair;
+import java.util.List;
+
+import com.yoti.api.client.Media;
+import com.yoti.api.client.docs.DocScanException;
+import com.yoti.api.client.docs.session.create.CreateSessionResult;
+import com.yoti.api.client.docs.session.create.SessionSpec;
+import com.yoti.api.client.docs.session.create.SimpleCreateSessionResult;
+import com.yoti.api.client.docs.session.retrieve.DocScanSession;
+import com.yoti.api.client.spi.remote.MediaValue;
+import com.yoti.api.client.spi.remote.call.*;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import com.yoti.api.client.spi.remote.call.factory.UnsignedPathFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Service built to handle the interactions between the client and
+ * Doc Scan API's.
+ *
+ * @since 2.7
+ */
+public final class DocScanService {
+
+    public static DocScanService newInstance() {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.FAIL_ON_EMPTY_BEANS, false);
+        objectMapper.setSerializationInclusion(JsonInclude.Include.NON_NULL);
+
+        return new DocScanService(new UnsignedPathFactory(), objectMapper);
+    }
+
+    private static final Logger LOG = LoggerFactory.getLogger(DocScanService.class);
+
+    private final UnsignedPathFactory unsignedPathFactory;
+    private final ObjectMapper objectMapper;
+
+    private final String apiUrl;
+
+    private DocScanService(UnsignedPathFactory pathFactory, ObjectMapper objectMapper) {
+        this.unsignedPathFactory = pathFactory;
+        this.objectMapper = objectMapper;
+
+        apiUrl = System.getProperty(PROPERTY_YOTI_DOCS_URL, DEFAULT_YOTI_DOCS_URL);
+    }
+
+    /**
+     * Performs a POST request using the supplied session specification to create
+     * a session, returning information about the newly created session if successful
+     *
+     * @param appId the app ID
+     * @param keyPair the {@code KeyPair}
+     * @param sessionSpec the {@code SessionSpec}
+     * @return the session creation result
+     * @throws DocScanException if there was an error whilst performing
+     *                          the request to Doc Scan systems
+     */
+    public CreateSessionResult createSession(String appId, KeyPair keyPair, SessionSpec sessionSpec) throws DocScanException {
```

Comment:
We call this method with a parameter named "sdkId", and assuming that this is correct, can we change this to sdkId too? 
```suggestion
    public CreateSessionResult createSession(String sdkId, KeyPair keyPair, SessionSpec sessionSpec) throws DocScanException {
```